lib/pure/strutils.nim

```diff
@@ -1370,17 +1371,69 @@ when not defined(js):
   proc c_sprintf(buf, frmt: cstring): cint {.header: "<stdio.h>",
                                      importc: "sprintf", varargs, noSideEffect.}
 
+# Forward declaration due to formatEng's use of ffDecimal formatting
+proc formatEng*(f: BiggestFloat,
+                precision: range[0..32] = 10,
+                decimalSep = '.',
+                trim: bool = true,
```

Comment:
As an engineer, the floating point / integer distinction isn't that important to me (unless I'm programming obviously).  The main reason I implemented trim was to avoid precision = 10 meaning you got 0.0000000000 out of the floating point print functions.  I could modify the meaning of trim such that it never removed the decimal place so that it would print 0.0 or 0.34 or 0.3623 but never 0.00 or 0.340 or 0.3623000000.  Is that better?